src/utils/http-date.ts

```diff
@@ -0,0 +1,295 @@
+/**
```

Comment:
This is grabbed straight from undici, do we really need to have a copy of this in our code? Is there a 3rd party library we can use? I really want to avoid us from maintaining such kind of code.

Reply:
@nodejs/undici I hope you don't mind the ping, but is this date parsing logic something y'all would consider exposing?

Reply:
> This is grabbed straight from undici, do we really need to have a copy of this in our code? Is there a 3rd party library we can use? I really want to avoid us from maintaining such kind of code.

Why not? We have parsing logic already for other things such as the `range` header